backend/mcp_backend/management/mcp_manager.py

```
@@ -0,0 +1,623 @@
+                # Process has terminated, remove it
+                del self.server_processes[name]
+
+        config = self.server_configs[name]
+
+        try:
+            # Check if this is an HTTP server (has --http in command)
+            is_http_server = "--http" in config["command"]
+
+            if is_http_server:
+                # Start HTTP server - no stdio pipes needed
+                process = subprocess.Popen(
```

Ruff (0.14.7)

`subprocess` call: check for execution of untrusted input

(S603)